src/dune_rules/dune_env.ml

```diff
@@ -19,7 +19,28 @@ module Stanza = struct
       Option.map since ~f:(fun since ->
           Dune_lang.Syntax.since Menhir_stanza.syntax since)
     in
-    Ordered_set_lang.Unexpanded.field "menhir_flags" ?check
+    located (Ordered_set_lang.Unexpanded.field "menhir_flags" ?check)
+    >>| fun (loc, flags) ->
+    let () =
+      Ordered_set_lang.Unexpanded.fold_strings flags ~init:()
+        ~f:(fun _pos sw acc ->
+          match String_with_vars.text_only sw with
+          | None -> acc
+          | Some text ->
+            if String.equal text "--explain" then
```

Comment:
Do we have the same error for cmly? If no, I would suggest to keep it uniform and add a similar error there (can be done in a separate PR).

Reply:
Also, this check should be guarded by 2.2

Reply:
I've made a patch making this uniform that I'll submit after this is merged.